pkg/sql/catalog/descriptor.go

```diff
@@ -170,42 +200,129 @@ type Descriptor interface {
+	// configured. If so, GetRegionConfig can be used.
 	IsMultiRegion() bool
+	// PrimaryRegionName returns the primary region for a multi-region database.
 	PrimaryRegionName() (descpb.RegionName, error)
+	// MultiRegionEnumID returns the ID of the multi-region enum if the database
+	// is a multi-region database, and an error otherwise.
 	MultiRegionEnumID() (descpb.ID, error)
+	// ForEachSchemaInfo iterates f over each schema info mapping in the descriptor.
+	// iterutil.StopIteration is supported.
 	ForEachSchemaInfo(func(id descpb.ID, name string, isDropped bool) error) error
+	// GetSchemaID returns the ID in the schema mapping entry for the
+	// given name, 0 otherwise.
 	GetSchemaID(name string) descpb.ID
+	// GetNonDroppedSchemaName returns the name in the schema mapping entry for the
+	// given ID, if it's not marked as dropped, empty string otherwise.
 	GetNonDroppedSchemaName(schemaID descpb.ID) string
+	// GetDefaultPrivilegeDescriptor returns the default privileges for this
+	// database.
 	GetDefaultPrivilegeDescriptor() DefaultPrivilegeDescriptor
 }
 
 // TableDescriptor is an interface around the table descriptor types.
 type TableDescriptor interface {
 	Descriptor
 
+	// TableDesc returns the backing protobuf for this database.
 	TableDesc() *descpb.TableDescriptor
 
+	// GetState returns the raw state of this descriptor. This information is
+	// mimicked by the Dropped(), Added(), etc information in Descriptor.
 	GetState() descpb.DescriptorState
+
+	// IsTable returns true if the TableDescriptor actually describes a
+	// Table resource, as opposed to a different resource (like a View).
+	IsTable() bool
+	// IsView returns true if the TableDescriptor actually describes a
+	// View resource rather than a Table.
+	IsView() bool
+	// IsSequence returns true if the TableDescriptor actually describes a
+	// Sequence resource rather than a Table.
+	IsSequence() bool
+	// IsTemporary returns true if this is a temporary table.
+	IsTemporary() bool
+	// IsVirtualTable returns true if the TableDescriptor describes a
+	// virtual Table (like the information_schema tables) and thus doesn't
+	// need to be physically stored.
+	IsVirtualTable() bool
+	// IsPhysicalTable returns true if the TableDescriptor actually describes a
+	// physical Table that needs to be stored in the kv layer, as opposed to a
+	// different resource like a view or a virtual table. Physical tables have
+	// primary keys, column families, and indexes (unlike virtual tables).
+	// Sequences count as physical tables because their values are stored in
+	// the KV layer.
+	IsPhysicalTable() bool
+	// IsInterleaved returns true if any part of this this table is interleaved with
+	// another table's data.
+	IsInterleaved() bool
+	// MaterializedView returns whether or not this TableDescriptor is a
+	// MaterializedView.
+	MaterializedView() bool
+	// IsAs returns true if the TableDescriptor describes a Table with an As
+	// source.
```

Comment:
Nit: "an As source" can be weird to read for the uninitiated (who in turn stands to benefit most from this comment) so this comment might benefit from mentioning CREATE TABLE ... AS SELECT as an instance of this.